### PR TITLE
Update testcafe-browser-natives to v0.13.0 (close #577)

### LINKED ISSUE #577: Update testcafe-browser-natives to v0.12.0
https://github.com/DevExpress/testcafe-browser-natives/pull/68

### PR DESCRIPTION
**Pull request by helen-dikareva**
/cc @AlexanderMoskovkin  @AndreyBelym

### CODE REVIEW

**Comment by AlexanderMoskovkin** [comment]
lgtm

File: test/functional/fixtures/api/es-next/resize-window/test.js [hunk 1]
```diff
@@ -7,6 +8,13 @@ if (!config.isTravisTask) {
             it('Should resize the window', function () {
                 return runTests('./testcafe-fixtures/resize-window-test.js', 'Resize the window');
             });
+
+            it('Should fail when a js-error appears during resizeWindow execution', function () {
+                return runTests('./testcafe-fixtures/resize-window-error-test.js', 'Resize the window leads to js-error', { shouldFail: true })
+                    .catch(function (errs) {
+                        errorInEachBrowserContains(errs, 'Error on page "http://localhost:3000/api/es-next/resize-window/pages/error-page.html":', 0);
```
Comment (AlexanderMoskovkin) [inline test/functional/fixtures/api/es-next/resize-window/test.js]: need to check callsite

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
FPR

**Comment by helen-dikareva** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
In general it's lgtm for me. I have some doubts but can't find a better way.
/cc @inikulin Look at this if you have some time

File: src/test-run/browser-manipulation-manager.js [hunk 2]
```diff
@@ -45,7 +43,9 @@ export default class BrowserManipulationManager {
     }
 
     static async resizeWindowToFitDevice (windowId, currentWidth, currentHeight, device, portrait) {
-        return await resizeWindow(windowId, currentWidth, currentHeight, device,
-            portrait ? PORTRAIT_ORIENTATION : LANDSCAPE_ORIENTATION);
+        var { landscapeWidth, portraitWidth } = getDeviceSize(device);
+
+        return await resizeWindow(windowId, currentWidth, currentHeight,
+            portrait ? portraitWidth : landscapeWidth, portrait ? landscapeWidth : portraitWidth);
```
Comment (inikulin) [inline src/test-run/browser-manipulation-manager.js]: Let's refactor ternaries to vars

File: src/test-run/commands/index.js [hunk 3]
```diff
@@ -622,7 +623,8 @@ export function createCommandFromObject (obj) {
 }
 
 export function isCommandRejectableByPageError (command) {
-    return !isObservationCommand(command) && !isWindowManipulationCommand(command) && !isServiceCommand(command);
+    return !isObservationCommand(command) && !isWindowManipulationCommand(command) && !isServiceCommand(command) &&
+           !(command.type === TYPE.prepareBrowserManipulation && command.manipulationType === TYPE.takeScreenshot);
```
Comment (inikulin) [inline src/test-run/commands/index.js]: I find it hard to parse this. Can we refactor it to make it clear when command rejectable and when not? BTW, in some places we have `browser manipulation` and in other `window manipulation`, can we come up with consistent terminology?

File: src/test-run/index.js [hunk 4]
```diff
@@ -166,7 +166,8 @@ export default class TestRun extends Session {
     }
 
     _removeAllNonServiceTasks () {
-        this.driverTaskQueue = this.driverTaskQueue.filter(driverTask => isServiceCommand(driverTask.command));
+        this.driverTaskQueue          = this.driverTaskQueue.filter(driverTask => isServiceCommand(driverTask.command));
+        this.browserManipulationQueue = this.browserManipulationQueue.filter(driverTask => isServiceCommand(driverTask.command));
```
Comment (inikulin) [inline src/test-run/index.js]: It should be `manipulationTask` I guess

File: src/test-run/commands/index.js [hunk 5]
```diff
@@ -504,8 +504,9 @@ export class TakeScreenshotOnFailCommand {
 }
 
 export class PrepareBrowserManipulationCommand {
-    constructor () {
-        this.type = TYPE.prepareBrowserManipulation;
+    constructor (manipulationType) {
+        this.type             = TYPE.prepareBrowserManipulation;
+        this.manipulationType = manipulationType;
```
Comment (inikulin) [inline src/test-run/commands/index.js]: `manipulationCommandType`?

**Comment by inikulin** [comment]
\r-

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by AndreyBelym** [comment]
lgtm

**Comment by helen-dikareva** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
the `testcafe-browser-natives@0.13.0` was published yesterday. Let's migrate to it in this PR. It contains a breaking change (https://github.com/DevExpress/testcafe-browser-natives/pull/69)

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by helen-dikareva** [comment]
@inikulin review please instead of @AndreyBelym

**Comment by AlexanderMoskovkin** [comment]
rebase required

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by AndreyBelym** [comment]
lgtm, sorry for inconveniences, the graduation is just eating all my time :sob:

**Comment by inikulin** [comment]
lgtm
